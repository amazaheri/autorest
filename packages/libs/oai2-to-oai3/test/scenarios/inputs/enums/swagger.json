{
  "swagger": "2.0",
  "info": {
    "version": "1.0",
    "title": "test",
    "description": "Test"
  },
  "produces": ["application/json"],
  "consumes": ["application/json"],
  "paths": {},
  "definitions": {
    "MyStringEnum": {
      "type": "object",
      "enum": ["foo", "bar"]
    },
    "MyNumberEnum": {
      "type": "object",
      "enum": [7, 998, 12310]
    },
    "MyObjectEnum": {
      "type": "object",
      "enum": [
        {
          "$ref": "#/definitions/FrequencySingle"
        }
      ]
    },
    "MyEnumValue": {
      "type": "object",
      "properties": {
        "age": {
          "type": "integer"
        }
      }
    }
  },
  "parameters": {}
}
