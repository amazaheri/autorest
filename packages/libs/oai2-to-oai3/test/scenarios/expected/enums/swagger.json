{
  "servers": [],
  "openapi": "3.0.0",
  "info": {
    "version": "1.0",
    "title": "test",
    "description": "Test"
  },
  "paths": {},
  "components": {
    "schemas": {
      "MyStringEnum": {
        "type": "object",
        "enum": [
          "foo",
          "bar"
        ]
      },
      "MyNumberEnum": {
        "type": "object",
        "enum": [
          7,
          998,
          12310
        ]
      },
      "MyObjectEnum": {
        "type": "object",
        "enum": [
          {
            "$ref": "#/components/schemas/FrequencySingle"
          }
        ]
      },
      "MyEnumValue": {
        "type": "object",
        "properties": {
          "age": {
            "type": "integer"
          }
        }
      }
    }
  }
}